src/OrbitGl/SamplingReport.cpp

```diff
@@ -20,11 +20,11 @@ using orbit_client_data::ThreadSampleData;
 using orbit_client_protos::CallstackInfo;
 
 SamplingReport::SamplingReport(
-    OrbitApp* app, PostProcessedSamplingData post_processed_sampling_data,
-    absl::flat_hash_map<uint64_t, std::shared_ptr<CallstackInfo>> unique_callstacks,
+    OrbitApp* app, const orbit_client_data::CallstackData* callstack_data,
+    const orbit_client_data::PostProcessedSamplingData* post_processed_sampling_data,
     bool has_summary)
-    : post_processed_sampling_data_{std::move(post_processed_sampling_data)},
-      unique_callstacks_{std::move(unique_callstacks)},
+    : callstack_data_{callstack_data},
```

Comment:
Can those be `nullptr`? If not, I'd add a `CHECK` here.

Reply:
They can't, added checks.